apache/couchdb-fauxton
# Fix sidebar expand/collapse when design doc name has special chars

## Overview

Fixes two issues:
 - When a design doc name has special chars (e.g. $, #), the item on the side bar was showing the encoded value (e.g. `my-$doc` is displayed as `my-%23doc`).
 - Clicking the design doc item would throw and the item was not expanded.

## Testing recommendations

- In a database click to create a new view. Name the design doc `my-design-$-#doc`, and the view `my-$-#view`.
   - Both names should display correctly on the sidebar
   - Click the design doc to expand/collapse it
   - Make sure after clicking the item is highlighted (text turns red)
- Load the view using the URL
   - Type the URL directly in the browser (e.g. `http://<hsot>:8001/#/database/stuff/_design/my-design-%24-%23doc/_view/my-%24-%23view`) and press Enter.

## GitHub issue number

https://github.com/cloudant/dashboard/issues/822
